dojo/tools/dockerbench/parser.py

```diff
@@ -0,0 +1,115 @@
+import json
+
+from dojo.models import Finding
+
+
+class DockerBenchParser(object):
+
+    def get_scan_types(self):
+        return ["docker-bench-security Scan"]
+
+    def get_label_for_scan_types(self, scan_type):
+        return scan_type  # no custom label for now
+
+    def get_description_for_scan_types(self, scan_type):
+        return "Import JSON reports of Docker CIS benchmark scans."
+
+    def get_findings(self, json_output, test):
+        tree = json.load(json_output)
+
+        return get_tests(tree, test)
+
+
+def get_tests(tree, test):
+    items_from_tests = []
+
+    description = ''
+    if 'id' in tree:
+        description += tree['id'] + " "
+    if 'text' in tree:
+        description += tree['text']
+    description += '\n'
+
+    for node in tree['tests']:
+        items_from_results = get_results(node, test, description)
+        items_from_tests += items_from_results
+
+    return list(items_from_tests)
+
+
+def get_results(tree, test, description):
+    items_from_results = []
+
+    if 'section' in tree:
+        description += tree['section'] + ' '
+    if 'desc' in tree:
+        description += tree['desc']
+    description += '\n'
+
+    for node in tree['results']:
+        item = get_item(node, test, description)
+        if item:
+            items_from_results.append(item)
+
+    return list(items_from_results)
+
+
+def get_item(vuln, test, description):
+
+    status = vuln.get('result', None)
+    reason = vuln.get('desc', None)
+
+    if status is None:
+        return None
+
+    # kube-bench doesn't define severities. So we use the status to define the severity
+    if status.upper() == 'FAIL':
+        severity = 'Critical'
+    elif status.upper() == 'WARN' and '(Manual)' not in reason:
+        severity = 'High'
+    elif status.upper() == 'INFO' and '(Manual)' not in reason:
+        severity = 'Low'
+    elif status.upper() == 'NOTE' and '(Manual)' not in reason:
+        severity = 'Info'
+    else:
+        return None
+
+    test_number = vuln.get('id', 'Test number not found')
```

Comment:
I think we should consider that `id` is require for this parser and organize the code like this.

Reply:
I agree, but I think code should not crash when there no id (user could edit file manually),
i tried to fix and now:
vuln_id_from_tool will be id or None, 
and title will be "<id> - <desc>", or "No test number - <desc>"

Reply:
this is good, will check again the code